src/marks/line.ts

```diff
@@ -1,219 +1,127 @@
+  _bufferManager.setResolution((ctx as any)._uniforms.resolution);
+  _bufferManager.setOffset([vb.x1, vb.y1]);
+
+
+  if(this.simpleLine === true) {
+    const uniformBindGroup = Renderer.createUniformBindGroup("S" + drawName, device, _pipeline2, _bufferManager.createUniformBuffer())
+    const attributes = createAttributes(items);
+    const instanceBuffer = _bufferManager.createInstanceBuffer(attributes);
+
+    Renderer.bundle2(device, _pipeline2, [6, items.length - 1], [instanceBuffer], [uniformBindGroup]);  
+  } else {
+    const uniformBindGroup = Renderer.createUniformBindGroup(drawName, device, _pipeline, _bufferManager.createUniformBuffer())
+    const pointDatas = createPointDatas(items);
+    const pointPositionBuffer = _bufferManager.createBuffer(drawName + ' Point Position Buffer', pointDatas.pos, GPUBufferUsage.STORAGE | GPUBufferUsage.COPY_DST);
+    const pointColorBuffer = _bufferManager.createBuffer(drawName + ' Point Color Buffer', pointDatas.colors, GPUBufferUsage.STORAGE | GPUBufferUsage.COPY_DST);
+    const pointWidthBuffer = _bufferManager.createBuffer(drawName + ' Point Width Buffer', pointDatas.widths, GPUBufferUsage.STORAGE | GPUBufferUsage.COPY_DST);
+    const pointBindGroup = Renderer.createBindGroup(drawName, device, _pipeline, [pointPositionBuffer, pointColorBuffer, pointWidthBuffer], null, 1);
+
+    Renderer.bundle2(device, _pipeline,  [6, items.length - 1], [], [uniformBindGroup, pointBindGroup]);
   }
 
-  const positionBuffer = createBuffer(device, Float32Array.from(positions), GPUBufferUsage.VERTEX);
-
-  const uniforms = Float32Array.from([...this._uniforms.resolution, vb.x1, vb.y1]);
-  const uniformBuffer = createBuffer(device, uniforms, GPUBufferUsage.UNIFORM | GPUBufferUsage.COPY_DST);
-  const uniformBindGroup = device.createBindGroup({
-    label: 'Line Uniform Bind Group',
-    layout: pipeline.getBindGroupLayout(0),
-    entries: [
-      {
-        binding: 0,
-        resource: {
-          buffer: uniformBuffer,
-        },
-      },
-    ],
-  });
-
-  const commandEncoder = device.createCommandEncoder();
-  //@ts-ignore
-  const textureView = ctx.getCurrentTexture().createView();
-  const renderPassDescriptor = {
-    label: 'Line Render Pass Descriptor',
-    colorAttachments: [
-      {
-        view: textureView,
-        loadOp: 'load',
-        storeOp: 'store',
-      },
-    ],
-  };
-
-  //@ts-ignore
-  const passEncoder = commandEncoder.beginRenderPass(renderPassDescriptor);
-  passEncoder.setPipeline(pipeline);
-  passEncoder.setBindGroup(0, uniformBindGroup);
-  passEncoder.setVertexBuffer(0, positionBuffer);
-  // 6 because our lines are made of quads
-  passEncoder.draw(itemCount * 6, 1);
-  passEncoder.end();
-  device.queue.submit([commandEncoder.finish()]);
 }
 
-function pick(context, scene, x, y, gx, gy) {
-  return null;
+function createAttributes(items: SceneItem[]) {
+  const lines = items as SceneLine[];
+  const result = new Float32Array((items.length - 1) * 9);
```

Comment:
We should be storing `9` as a constant where we define the vertex descriptor rather than as a magic number. Same for all other marks.